src/test/java/org/opensearch/neuralsearch/processor/RRFSearchIT.java

```diff
@@ -0,0 +1,171 @@
+        Settings indexSettings = Settings.builder().put("index.knn", true).put("default_pipeline", ingestPipelineName).build();
+        String indexMappings = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("properties")
+            .startObject("id")
+            .field("type", "text")
+            .endObject()
+            .startObject("passage_embedding")
+            .field("type", "knn_vector")
+            .field("dimension", "768")
+            .startObject("method")
+            .field("engine", "lucene")
+            .field("space_type", "l2")
+            .field("name", "hnsw")
+            .endObject()
+            .endObject()
+            .startObject("text")
+            .field("type", "text")
+            .endObject()
+            .endObject()
+            .endObject()
+            .toString();
+        // Removes the {} around the string, since they are already included with createIndex
+        indexMappings = indexMappings.substring(1, indexMappings.length() - 1);
+        String indexName = "rrf-index";
+        createIndex(indexName, indexSettings, indexMappings, null);
+        addRRFDocuments();
+        createDefaultRRFSearchPipeline();
+
+        Map<String, Object> results = searchRRF(modelId);
+        Map<String, Object> hits = (Map<String, Object>) results.get("hits");
+        ArrayList<HashMap<String, Object>> hitsList = (ArrayList<HashMap<String, Object>>) hits.get("hits");
+        assertEquals(3, hitsList.size());
+        assertEquals(0.016393442, (Double) hitsList.getFirst().get("_score"), DELTA_FOR_SCORE_ASSERTION);
+        assertEquals(0.016129032, (Double) hitsList.get(1).get("_score"), DELTA_FOR_SCORE_ASSERTION);
+        assertEquals(0.015873017, (Double) hitsList.getLast().get("_score"), DELTA_FOR_SCORE_ASSERTION);
+    }
+
+    @SneakyThrows
+    private void addRRFDocuments() {
+        addRRFDocument(
+            "A West Virginia university women 's basketball team , officials , and a small gathering of fans are in a West Virginia arena .",
+            "4319130149.jpg"
+        );
+        addRRFDocument("A wild animal races across an uncut field with a minimal amount of trees .", "1775029934.jpg");
+        addRRFDocument(
+            "People line the stands which advertise Freemont 's orthopedics , a cowboy rides a light brown bucking bronco .",
+            "2664027527.jpg"
+        );
+        addRRFDocument("A man who is riding a wild horse in the rodeo is very near to falling off .", "4427058951.jpg");
+        addRRFDocument("A rodeo cowboy , wearing a cowboy hat , is being thrown off of a wild white horse .", "2691147709.jpg");
+    }
+
+    @SneakyThrows
+    private void addRRFDocument(String description, String imageText) {
+        addDocument(RRF_INDEX_NAME, String.valueOf(currentDoc++), "text", description, "image_text", imageText);
+    }
+
+    @SneakyThrows
+    private void createDefaultRRFSearchPipeline() {
```

Comment:
check if we can extend the existing method in [base class](https://github.com/opensearch-project/neural-search/blob/main/src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java) or at least move this method there

Reply:
I was trying to do that, but it requires the method in the base class requires a normalization input and the rrf pipeline doesn't have one.  Unless there's something I'm missing, we should just move it to the base class.